### PR TITLE
Add a fallback route for the back end

### LINKED ISSUE #7024: Not existing `/contao` URL generates front end 401 page
### Affected version(s)

4.13, 5.3

### Description

If you access a URL with the `/contao` (`%contao.backend.route_prefix%`) prefix, but the URL does not exist in the router, the _front end_ 401 page will be generated. For example: https://demo.contao.org/contao/foobar

I expected either the back end login to be shown (if you are not logged in) or the front end 404 page but not its 401 page.

### PR DESCRIPTION
**Pull request by aschempp**
Fixes https://github.com/contao/contao/issues/7024

This will render the back end error screen, not the Symfony 404 message. I tried using a controller that throws an `NotFoundHttpException`, but that will result in the exact same template. Because our pretty error screen always handles backend exceptions.

Bildschirmfoto 2024-04-04 um 21 27 22

### CODE REVIEW

**Review by fritzmg — DISMISSED** [review]

**Comment by fritzmg** [comment]
I can also confirm that a request to `/contao/foobar` first redirects to `/contao/login` (with `redirect=…%2Fcontao%2Ffoobar`), just like any other regular Contao back end route.

**Review by ausi — DISMISSED** [review]

**Comment by leofeyer** [comment]
Do we want to have a nicer, perhaps even translatable error message?

**Comment by aschempp** [comment]
A 404 error page design only for the back end, where noone is supposed to "randomly" try URLs or ever access something that does not exist?

**Comment by leofeyer** [comment]
People are constantly using deep links in the back end, so it is not unlikely that the page will be seen a lot.

**Comment by aschempp** [comment]
But deep links would never point to a 404? Anyway, I think this PR is ready to merge and fixes a bug. Adding a 404 page can be done if someone feels like adding that feature 😊

File: core-bundle/src/Resources/views/Error/layout.html.twig [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% trans_default_domain 'contao_exception' %}
 <!DOCTYPE html>
-<html lang="{{ language }}">
+<html lang="{{ language|default(app.request.locale) }}">
```
Comment (leofeyer) [inline core-bundle/src/Resources/views/Error/layout.html.twig:3]: This seems unrelated?

**Review by leofeyer — DISMISSED** [review]

**Review by leofeyer — DISMISSED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
@fritzmg This pull request was merged into Contao 4.13, however, the fix in #7222 is only for Contao 5.3. Don‘t we have to fix this in Contao 4.13, too?

**Comment by fritzmg** [comment]
Yes, I missed that the follow-ups were only for 5.3.
